connector-email/src/main/java/com/redhat/cloud/notifications/connector/email/EmailRouteBuilder.java

```diff
@@ -303,4 +307,31 @@ private Endpoint setUpITEndpoint() throws Exception {
 
         return httpComponent.createEndpoint(String.format("https:%s", this.emailConnectorConfig.getItUserServiceURL()));
     }
+
+    /**
+     * Creates the endpoint for the BOP service. It makes Apache Camel trust
+     * BOP service's certificate.
+     * @return the created endpoint.
+     */
+    protected Endpoint setUpBOPEndpoint() {
+        final TrustManagersParameters trustManagersParameters = new TrustManagersParameters();
+        trustManagersParameters.setTrustManager(new BOPTrustManager());
+
+        final SSLContextParameters sslContextParameters = new SSLContextParameters();
+        sslContextParameters.setTrustManagers(trustManagersParameters);
+
+        // Remove the schema from the url to avoid the "ResolveEndpointFailedException".
+        final String fullURL = this.emailConnectorConfig.getBopURL();
+        final String noProtocolURL;
+        if (fullURL.startsWith("https")) {
+            noProtocolURL = fullURL.replace("https://", "");
+        } else {
+            noProtocolURL = fullURL.replace("http://", "");
+        }
+
+        return https(noProtocolURL)
+            .sslContextParameters(sslContextParameters)
+            .x509HostnameVerifier(NoopHostnameVerifier.INSTANCE)
+            .resolve(this.getContext());
```

Comment:
What does this line do exactly?

Reply:
I understood it as "it resolves the whole endpoint builder as an endpoint": https://github.com/apache/camel/blob/7d76dc29d6ae82eb2cd92762a31883714d77e5f7/core/camel-api/src/main/java/org/apache/camel/EndpointProducerResolver.java#L24-L31

But it is true that it is not very clear... Maybe I can simply change it.